iac-extensions/spring-config/src/main/java/org/sonar/iac/springconfig/tree/impl/TupleImpl.java

```diff
@@ -0,0 +1,51 @@
+/*
+ * SonarQube IaC Plugin
+ * Copyright (C) 2021-2024 SonarSource SA
+ * mailto:info AT sonarsource DOT com
+ *
+ * This program is free software; you can redistribute it and/or
+ * modify it under the terms of the GNU Lesser General Public
+ * License as published by the Free Software Foundation; either
+ * version 3 of the License, or (at your option) any later version.
+ *
+ * This program is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the GNU
+ * Lesser General Public License for more details.
+ *
+ * You should have received a copy of the GNU Lesser General Public License
+ * along with this program; if not, write to the Free Software Foundation,
+ * Inc., 51 Franklin Street, Fifth Floor, Boston, MA  02110-1301, USA.
+ */
+package org.sonar.iac.springconfig.tree.impl;
+
+import java.util.Arrays;
+import java.util.List;
+import org.sonar.iac.common.api.tree.Tree;
+import org.sonar.iac.springconfig.tree.api.Scalar;
+import org.sonar.iac.springconfig.tree.api.Tuple;
+
+public class TupleImpl extends AbstractSpringConfigImpl implements Tuple {
+  private final Scalar key;
+  private final Scalar value;
+
+  public TupleImpl(Scalar key, Scalar value) {
+    this.key = key;
+    this.value = value;
+  }
+
+  @Override
+  public List<Tree> children() {
+    return Arrays.asList(key, value);
```

Comment:
`List.of` to have an immutable collection?

Reply:
OK, that may make sense :thinking: 
However, then we need to annotate potentially nullable parameters. I believe in a valid properties file `key` will never be null, but the `value` may. OTOH, we can create a tuple with an artificial empty SyntaxToken in such case.